pkg/recommendation/factory.go

```diff
@@ -34,36 +34,46 @@ func NewFactory(logger logrus.FieldLogger, dynamicCli dynamic.Interface) *Factor
 }
 
 // NewForSources merges recommendation options from multiple sources, and creates a new Set.
-func (f *Factory) NewForSources(sources map[string]config.Sources) Set {
+func (f *Factory) NewForSources(sources map[string]config.Sources, mapKeyOrder []string) Set {
```

Comment:
IMO it can be simplified a bit:
```go

// NewForSources merges recommendation options from multiple sources, and creates a new Set.
func (f *Factory) NewForSources(sources map[string]config.Sources, mapKeyOrder []string) Set {
	
	rec := config.Recommendations{}
	for _, key := range mapKeyOrder {
		source, exists := sources[key]
		if !exists {
			continue
		}

		sourceRec := source.Kubernetes.Recommendations
		if sourceRec.Pod.LabelsSet != nil {
			rec.Pod.LabelsSet = sourceRec.Pod.LabelsSet
		}
		if sourceRec.Pod.NoLatestImageTag != nil {
			rec.Pod.NoLatestImageTag = sourceRec.Pod.NoLatestImageTag
		}
		if sourceRec.Ingress.BackendServiceValid != nil {
			rec.Ingress.BackendServiceValid = sourceRec.Ingress.BackendServiceValid
		}
		if sourceRec.Ingress.TLSSecretValid != nil {
			rec.Ingress.TLSSecretValid = sourceRec.Ingress.TLSSecretValid
		}
	}

	set := f.enableRecommendationIfShould(rec)
	return newRecommendationsSet(f.logger, set)
}


func (f *Factory) enableRecommendationIfShould(rec config.Recommendations) map[string]Recommendation {
	var list []Recommendation
	if isTrue(rec.Pod.LabelsSet) {
		list = append(list, NewPodLabelsSet())
	}

	if isTrue(rec.Pod.NoLatestImageTag) {
		list = append(list, NewPodNoLatestImageTag())

	}

	if isTrue(rec.Ingress.BackendServiceValid) {
		list = append(list, NewIngressBackendServiceValid(f.dynamicCli))

	}

	if isTrue(rec.Ingress.TLSSecretValid) {
		list = append(list, NewIngressTLSSecretValid(f.dynamicCli))
	}

	out := map[string]Recommendation{}
	for _, rec := range list {
		out[rec.Name()] = rec
	}
	return out
}


func isTrue(in *bool) bool {
	if in == nil {
		return false
	}
	return *in
}
```

Currently, I see two problems:
- enableRecommendationIfShould - it's not only about "enable" or not it's about construct/destruct obj
- in a loop we create and destruct obj, one thing is GC but I don't care about that too much right now, for me it was just hard to follow the logic of it.